cli/factory.go

```diff
@@ -0,0 +1,346 @@
+func (f *Factory) BuildUpdater(ctx context.Context) (update.Updater, error) {
+	c, err := f.BuildClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build client: %w", err)
+	}
+
+	return update.NewUpdater(c), nil
+}
+
+// BuildSyncer builds a syncer.
+func (f *Factory) BuildSyncer(ctx context.Context) (sync.Syncer, error) {
+	c, err := f.BuildClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build client: %w", err)
+	}
+
+	return sync.NewSyncer(c), nil
+}
+
+// BuildLabeler builds a labeler.
+func (f *Factory) BuildLabeler(ctx context.Context) (label.Labeler, error) {
+	c, err := f.BuildClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build client: %w", err)
+	}
+
+	return label.NewLabeler(c), nil
+}
+
+// BuildInstaller builds an installer.
+func (f *Factory) BuildInstaller(ctx context.Context) (install.Installer, error) {
+	c, err := f.BuildClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build client: %w", err)
+	}
+
+	return install.NewInstaller(c), nil
+}
+
+// BuildInitializer builds an initializer.
+func (f *Factory) BuildInitializer(_ context.Context) (initialize.Initializer, error) {
+	return initialize.NewInitializer(f.cfg, f.cfgPath), nil
+}
+
+// BuildGetter builds a getter.
+func (f *Factory) BuildGetter(ctx context.Context) (get.Getter, error) {
+	c, err := f.BuildClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build client: %w", err)
+	}
+
+	printer, err := f.BuildPrinter(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build printer: %w", err)
+	}
+
+	return get.NewGetter(c, printer), nil
+}
+
+// BuildDeleter builds a deleter.
+func (f *Factory) BuildDeleter(ctx context.Context) (delete.Deleter, error) {
+	c, err := f.BuildClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build client: %w", err)
+	}
+
+	return delete.NewDeleter(c), nil
+}
+
+// BuildCopier builds a copier.
+func (f *Factory) BuildCopier(ctx context.Context) (copy.Copier, error) {
+	c, err := f.BuildClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build client: %w", err)
+	}
+
+	return copy.NewCopier(c), nil
+}
+
+// BuildApplier builds an applier.
+func (f *Factory) BuildApplier(ctx context.Context) (apply.Applier, error) {
+	c, err := f.BuildClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build client: %w", err)
+	}
+
+	return apply.NewApplier(c), nil
+}
+
+// BuildServer builds a server.
+func (f *Factory) BuildServer(ctx context.Context) (serve.Server, error) {
+	logger, err := f.BuildLogger(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build logger: %w", err)
+	}
+
+	st, err := f.BuildStore(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load store: %w", err)
+	}
+
+	tr, err := f.BuildTracer(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load tracer: %w", err)
+	}
+
+	return serve.NewServer(&f.cfg.Server, st, tr, logger), nil
+}
+
+// BuildLogger builds a logger.
```

Comment:
looked to me like we were building a new logger for everything, same with client, etc. think it might be worth at least updating the comment. because some of the builders do create a new version of `X` and some don't. like `BuildApplier` creates a new applier every time
```suggestion
// BuildLogger builds a logger, if it doesn't already exist.
```